andela/wolfsbane-backend
# #167750053 A user can log into the Barefoot Nomad application via email & password

#### Description of Tasks to be completed?
- setup controller for  user login
- setup route for  user login
- setup unit test for user login route

 #### How should this be manually tested?
- cd into the cloned directory
- make sure database connection is already set up
- run `npm install`
- ADD all environmental variable as included the **.env-example** to your **.env** file
- To run database migration  `npm run migrate`
- To test run `npm run test`
- run `npm run start:dev`
- To test with postman use `http://localhost:3000/api/v1/users/signin` with appropriate parameters

 #### Any background context you want to provide?
- Sample request:
```json
{
	"email": "email@gmail.com",
	"password": "password"
}
```

- Sample response :
 ```json
{
    "status": 200,
    "message": "Successfully Signed In",
    "data": {
        "firstName": "firstname",
        "lastName": "lastname",
        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJiZTI3NGRjYS0wMmQ0LTRmYjMtOThjYy1mYzJiZDVkYTY3ZGMiLCJpYXQiOjE1Njc1OTQ4NzMsImV4cCI6MTU2NzY4MTI3M30.SB8EPxDmHSqpBqizfVT_l8hVGO54D4-bl81DbeUH3Mk"
    }
}
```
 #### What are the relevant pivotal tracker stories?
 [#167750053](https://www.pivotaltracker.com/story/show/167750053)
#### Screenshots (if appropriate)
None
 #### Questions:
None
